gravitee-cockpit-connectors-test/src/test/java/io/gravitee/cockpit/connectors/test/TestInitializer.java

```diff
@@ -0,0 +1,215 @@
+    }
+
+    @Bean
+    WebSocketCockpitConnectorBeanFactory webSocketCockpitConnector() {
+        return new WebSocketCockpitConnectorBeanFactory();
+    }
+
+    protected WebClient webClient = WebClient.wrap(io.vertx.reactivex.core.Vertx.vertx().createHttpClient());
+
+    private Single<JsonArray> getAccounts(User user) {
+        return webClient
+            .getAbs(BASE_URL + "/management/accounts")
+            .basicAuthentication(user.getUsername(), user.getPassword())
+            .rxSend()
+            .map(HttpResponse::bodyAsJsonArray);
+    }
+
+    private Single<Certificate> getCertificate(User user, Account account) {
+        return webClient
+            .getAbs(BASE_URL + "/management/accounts/" + account.getId() + "/certificate")
+            .basicAuthentication(user.getUsername(), user.getPassword())
+            .rxSend()
+            .map(HttpResponse::bodyAsJsonObject)
+            .map(
+                objects ->
+                    new Certificate()
+                        .withId(objects.getString("id"))
+                        .withAccountId(objects.getString("accountId"))
+                        .withAlias(objects.getString("alias"))
+                        .withStorePass(objects.getString("storePass"))
+                        .withKeyPass(objects.getString("keyPass"))
+                        .withStoreType(objects.getString("storeType"))
+                        .withContent(objects.getString("content"))
+            );
+    }
+
+    private Single<CertificateAuthority> getCertificateAuthority(User user) {
+        return webClient
+            .getAbs(BASE_URL + "/management/certificateAuthority")
+            .basicAuthentication(user.getUsername(), user.getPassword())
+            .rxSend()
+            .map(HttpResponse::bodyAsJsonObject)
+            .map(
+                jsonCertificateAuthority ->
+                    new CertificateAuthority()
+                        .withAlias(jsonCertificateAuthority.getString("alias"))
+                        .withStorePass(jsonCertificateAuthority.getString("storePass"))
+                        .withStoreType(jsonCertificateAuthority.getString("storeType"))
+                        .withContent(jsonCertificateAuthority.getString("content"))
+                        .withExpireAt(ZonedDateTime.parse(jsonCertificateAuthority.getString("expireAt")))
+            );
+    }
+
+    private Single<Account> createAccount(User user) {
+        Faker faker = new Faker();
+        NewAccount newAccount = new NewAccount(faker.name().name(), UUID.randomUUID().toString(), faker.shakespeare().asYouLikeItQuote());
+
+        return webClient
+            .postAbs(BASE_URL + "/management/accounts")
+            .basicAuthentication(user.getUsername(), user.getPassword())
+            .rxSendJson(newAccount)
+            .map(HttpResponse::bodyAsJsonObject)
+            .map(json -> new Account().withId(json.getString("id")));
```

Comment:
Is it possible to parse the json response and get real data in account object ? Don't you think we might need the other fields at some point ?

Reply:
👍🏻 I used `Json.Decode` to autoparse the object with a minimal effort.